solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,679 @@
+  description = "Name of the Default ACL. If null, a name will be automatically generated."
+  type        = string
+  default     = null
+}
+
+variable "default_security_group_name" {
+  description = "Name of the Default Security Group. If null, a name will be automatically generated."
+  type        = string
+  default     = null
+}
+
+variable "default_routing_table_name" {
+  description = "Name of the Default Routing Table. If null, a name will be automatically generated."
+  type        = string
+  default     = null
+}
+
+##############################################################################
+# VPN Gateways
+##############################################################################
+
+variable "vpn_gateways" {
+  description = "List of VPN Gateways to create."
+  type = list(
+    object({
+      name           = string
+      vpc_name       = string
+      subnet_name    = string # Do not include prefix, use same name as in `var.subnets`
+      mode           = optional(string)
+      resource_group = optional(string)
+      access_tags    = optional(list(string), [])
+    })
+  )
+
+  default = []
+}
+
+##############################################################################
+# VPE Gateways
+##############################################################################
+
+variable "cloud_services" {
+  description = "The list of cloud services used to create endpoint gateways. If `vpe_name` is not specified in the list, VPE names are created in the format `<prefix>-<vpc_name>-<service_name>`."
+  type = set(object({
+    service_name                 = string
+    vpe_name                     = optional(string), # Full control on the VPE name. If not specified, the VPE name will be computed based on prefix, vpc name and service name.
+    allow_dns_resolution_binding = optional(bool, false)
+  }))
+  default = []
+}
+
+variable "cloud_service_by_crn" {
+  description = "The list of cloud service CRNs used to create endpoint gateways. Use this list to identify services that are not supported by service name in the `cloud_services` variable. For a list of supported services, see [VPE-enabled services](https://cloud.ibm.com/docs/vpc?topic=vpc-vpe-supported-services). If `service_name` is not specified, the CRN is used to find the name. If `vpe_name` is not specified in the list, VPE names are created in the format `<prefix>-<vpc_name>-<service_name>`."
+  type = set(
+    object({
+      crn                          = string
+      vpe_name                     = optional(string) # Full control on the VPE name. If not specified, the VPE name will be computed based on prefix, vpc name and service name.
+      service_name                 = optional(string) # Name of the service used to compute the name of the VPE. If not specified, the service name will be obtained from the crn.
+      allow_dns_resolution_binding = optional(bool, true)
+    })
+  )
+  default = []
+}
+
+variable "vpe_service_endpoints" {
```

Comment:
suggest to prefix this with `vpe_gateway_`

Reply:
updated